--- a/bolt/cmake/modules/AddBOLT.cmake
+++ b/bolt/cmake/modules/AddBOLT.cmake
@@ -16,7 +16,7 @@
     get_target_export_arg(${name} BOLT export_to_bolttargets)
     install(TARGETS ${name}
       ${export_to_bolttargets}
-      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      RUNTIME DESTINATION "${BOLT_TOOLS_INSTALL_DIR}"
       COMPONENT bolt)
 
     if(NOT LLVM_ENABLE_IDE)
--- a/clang-tools-extra/clang-tidy/tool/CMakeLists.txt
+++ b/clang-tools-extra/clang-tidy/tool/CMakeLists.txt
@@ -66,6 +66,6 @@
   DESTINATION "${CMAKE_INSTALL_DATADIR}/clang"
   COMPONENT clang-tidy)
 install(PROGRAMS run-clang-tidy.py
-  DESTINATION "${CMAKE_INSTALL_BINDIR}"
+  DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
   COMPONENT clang-tidy
   RENAME run-clang-tidy)
--- a/clang-tools-extra/modularize/CMakeLists.txt
+++ b/clang-tools-extra/modularize/CMakeLists.txt
@@ -28,5 +28,5 @@
   )
 
 install(TARGETS modularize
-        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        RUNTIME DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
         COMPONENT clang-extras)
--- a/clang/cmake/modules/AddClang.cmake
+++ b/clang/cmake/modules/AddClang.cmake
@@ -183,7 +183,7 @@
       get_target_export_arg(${name} Clang export_to_clangtargets)
       install(TARGETS ${name}
         ${export_to_clangtargets}
-        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        RUNTIME DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
         COMPONENT ${name})
 
       if (LLVM_ENABLE_PDB)
--- a/clang/tools/c-index-test/CMakeLists.txt
+++ b/clang/tools/c-index-test/CMakeLists.txt
@@ -49,7 +49,7 @@
     set_property(TARGET c-index-test APPEND PROPERTY INSTALL_RPATH
        "@executable_path/../../lib")
   else()
-    set(INSTALL_DESTINATION "${CMAKE_INSTALL_BINDIR}")
+    set(INSTALL_DESTINATION "${CLANG_TOOLS_INSTALL_DIR}")
   endif()
 
   install(TARGETS c-index-test
--- a/clang/tools/clang-format/CMakeLists.txt
+++ b/clang/tools/clang-format/CMakeLists.txt
@@ -36,7 +36,7 @@
   DESTINATION "${CMAKE_INSTALL_DATADIR}/clang"
   COMPONENT clang-format)
 install(PROGRAMS git-clang-format
-  DESTINATION "${CMAKE_INSTALL_BINDIR}"
+  DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
   COMPONENT clang-format)
 
 if (WIN32 AND NOT CYGWIN)
--- a/clang/tools/scan-build-py/CMakeLists.txt
+++ b/clang/tools/scan-build-py/CMakeLists.txt
@@ -46,7 +46,7 @@
                            ${CMAKE_BINARY_DIR}/bin/scan-build-py
                          DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/bin/scan-build)
       install (PROGRAMS "bin/scan-build"
-               DESTINATION "${CMAKE_INSTALL_BINDIR}"
+               DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
                RENAME scan-build-py
                COMPONENT scan-build-py)
       list(APPEND Depends ${CMAKE_BINARY_DIR}/bin/scan-build-py)
@@ -59,7 +59,7 @@
                            ${CMAKE_BINARY_DIR}/bin/
                          DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/bin/${BinFile})
       install(PROGRAMS bin/${BinFile}
-              DESTINATION "${CMAKE_INSTALL_BINDIR}"
+              DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
               COMPONENT scan-build-py)
       list(APPEND Depends ${CMAKE_BINARY_DIR}/bin/${BinFile})
     endif()
--- a/clang/tools/scan-build/CMakeLists.txt
+++ b/clang/tools/scan-build/CMakeLists.txt
@@ -47,7 +47,7 @@
                        DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/bin/${BinFile})
     list(APPEND Depends ${CMAKE_BINARY_DIR}/bin/${BinFile})
     install(PROGRAMS bin/${BinFile}
-            DESTINATION "${CMAKE_INSTALL_BINDIR}"
+            DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
             COMPONENT scan-build)
   endforeach()
 
--- a/clang/tools/scan-view/CMakeLists.txt
+++ b/clang/tools/scan-view/CMakeLists.txt
@@ -20,7 +20,7 @@
                        DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/bin/${BinFile})
     list(APPEND Depends ${CMAKE_BINARY_DIR}/bin/${BinFile})
     install(PROGRAMS bin/${BinFile}
-            DESTINATION "${CMAKE_INSTALL_BINDIR}"
+            DESTINATION "${CLANG_TOOLS_INSTALL_DIR}"
             COMPONENT scan-view)
   endforeach()
 
--- a/flang/cmake/modules/AddFlang.cmake
+++ b/flang/cmake/modules/AddFlang.cmake
@@ -122,7 +122,7 @@
     get_target_export_arg(${name} Flang export_to_flangtargets)
     install(TARGETS ${name}
       ${export_to_flangtargets}
-      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      RUNTIME DESTINATION "${FLANG_TOOLS_INSTALL_DIR}"
       COMPONENT ${name})
 
     if(NOT LLVM_ENABLE_IDE)
--- a/flang/tools/flang-driver/CMakeLists.txt
+++ b/flang/tools/flang-driver/CMakeLists.txt
@@ -41,10 +41,10 @@
 # Enable support for plugins, which need access to symbols from flang
 if(FLANG_PLUGIN_SUPPORT)
   set(LLVM_LIBRARY_OUTPUT_INTDIR ${LLVM_LIBRARY_DIR})
-  export_executable_symbols_for_plugins(flang)
+  export_executable_symbols_for_plugins(flang-new)
 endif()
 
-install(TARGETS flang DESTINATION "${CMAKE_INSTALL_BINDIR}")
+install(TARGETS flang-new DESTINATION "${FLANG_TOOLS_INSTALL_DIR}")
 
 # Keep "flang-new" as a symlink for backwards compatiblity. Remove once "flang"
 # is a widely adopted name.
--- a/lldb/cmake/modules/AddLLDB.cmake
+++ b/lldb/cmake/modules/AddLLDB.cmake
@@ -203,7 +203,7 @@
   endif()
 
   if(ARG_GENERATE_INSTALL)
-    set(install_dest bin)
+    set(install_dest "${LLVM_TOOLS_INSTALL_DIR}")
     if(ARG_INSTALL_PREFIX)
       set(install_dest ${ARG_INSTALL_PREFIX})
     endif()
