Subject: [PATCH] Disabling overloaded destructor for val<sstream>
---
Index: nautilus/include/nautilus/std/sstream.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/include/nautilus/std/sstream.h b/nautilus/include/nautilus/std/sstream.h
--- a/nautilus/include/nautilus/std/sstream.h	(revision 367eda593a725c2bef6c255de56e93dd71ece62c)
+++ b/nautilus/include/nautilus/std/sstream.h	(revision a8f83e3f44161e59d3be6d2ff10c90ba300dfcd0)
@@ -62,9 +62,9 @@
 		return val<std::basic_string<CharT, Traits>>(str_ptr);
 	}
 
-	~val() {
-		invoke(+[](std::basic_ostream<CharT, Traits>* ptr) -> void { delete ptr; }, this->stream);
-	}
+	// ~val() {
+	// 	invoke(+[](std::basic_ostream<CharT, Traits>* ptr) -> void { delete ptr; }, this->stream);
+	// }
 };
 
 typedef val<std::basic_stringstream<char>> stringstream;
