Subject: [PATCH] increase TypedValueRef from u16 to u32
---
Index: nautilus/include/nautilus/tracing/TypedValueRef.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/include/nautilus/tracing/TypedValueRef.hpp b/nautilus/include/nautilus/tracing/TypedValueRef.hpp
--- a/nautilus/include/nautilus/tracing/TypedValueRef.hpp	(revision c068e7a0f1ccc76d6e149cdf3004e83ec5f3c5bb)
+++ b/nautilus/include/nautilus/tracing/TypedValueRef.hpp	(revision a71bcabed738167def7a175607904934b8ad4bf0)
@@ -4,12 +4,12 @@
 #include "nautilus/tracing/Types.hpp"
 namespace nautilus::tracing {

-using ValueRef = uint16_t;
+using ValueRef = uint32_t;
 struct TypedValueRef {
 	ValueRef ref;
 	Type type;

-	TypedValueRef(uint16_t ref, Type type);
+	TypedValueRef(ValueRef ref, Type type);
 	TypedValueRef();

 	bool operator==(const TypedValueRef& rhs) const {
Index: nautilus/src/nautilus/common/TypedValueRefHolder.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/src/nautilus/common/TypedValueRefHolder.cpp b/nautilus/src/nautilus/common/TypedValueRefHolder.cpp
--- a/nautilus/src/nautilus/common/TypedValueRefHolder.cpp	(revision c068e7a0f1ccc76d6e149cdf3004e83ec5f3c5bb)
+++ b/nautilus/src/nautilus/common/TypedValueRefHolder.cpp	(revision a71bcabed738167def7a175607904934b8ad4bf0)
@@ -7,7 +7,7 @@
 TypedValueRef::TypedValueRef() : ref(0), type(Type::v) {
 }

-TypedValueRef::TypedValueRef(uint16_t ref, nautilus::Type type) : ref(ref), type(type) {
+TypedValueRef::TypedValueRef(ValueRef ref, nautilus::Type type) : ref(ref), type(type) {
 }

 TypedValueRefHolder::operator const TypedValueRef&() const {
Index: nautilus/src/nautilus/tracing/Block.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/src/nautilus/tracing/Block.hpp b/nautilus/src/nautilus/tracing/Block.hpp
--- a/nautilus/src/nautilus/tracing/Block.hpp	(revision c068e7a0f1ccc76d6e149cdf3004e83ec5f3c5bb)
+++ b/nautilus/src/nautilus/tracing/Block.hpp	(revision a71bcabed738167def7a175607904934b8ad4bf0)
@@ -10,7 +10,7 @@

 struct operation_identifier {
 	uint16_t blockIndex;
-	uint16_t operationIndex;
+	uint32_t operationIndex;
 };

 /**
Index: nautilus/src/nautilus/tracing/ExecutionTrace.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/src/nautilus/tracing/ExecutionTrace.cpp b/nautilus/src/nautilus/tracing/ExecutionTrace.cpp
--- a/nautilus/src/nautilus/tracing/ExecutionTrace.cpp	(revision c068e7a0f1ccc76d6e149cdf3004e83ec5f3c5bb)
+++ b/nautilus/src/nautilus/tracing/ExecutionTrace.cpp	(revision a71bcabed738167def7a175607904934b8ad4bf0)
@@ -214,7 +214,7 @@

 TypedValueRef& ExecutionTrace::setArgument(Type type, size_t index) {
 	++lastValueRef;
-	uint16_t argRef = index + 1;
+	ValueRef argRef = index + 1;
 	auto& arguments = blocks[0].arguments;
 	if (arguments.size() < argRef) {
 		arguments.resize(argRef);
@@ -228,13 +228,13 @@
 	return returnRefs;
 }

-uint16_t ExecutionTrace::getNextValueRef() {
+ValueRef ExecutionTrace::getNextValueRef() {
 	return ++lastValueRef;
 }

 operation_identifier ExecutionTrace::getNextOperationIdentifier() {
 	currentOperationIndex = getCurrentBlock().operations.size() - 1;
-	return {currentBlockIndex, (uint16_t) currentOperationIndex};
+	return {currentBlockIndex, currentOperationIndex};
 }

 void ExecutionTrace::resetExecution() {
Index: nautilus/src/nautilus/tracing/ExecutionTrace.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/src/nautilus/tracing/ExecutionTrace.hpp b/nautilus/src/nautilus/tracing/ExecutionTrace.hpp
--- a/nautilus/src/nautilus/tracing/ExecutionTrace.hpp	(revision c068e7a0f1ccc76d6e149cdf3004e83ec5f3c5bb)
+++ b/nautilus/src/nautilus/tracing/ExecutionTrace.hpp	(revision a71bcabed738167def7a175607904934b8ad4bf0)
@@ -169,9 +169,9 @@

 	/**
 	 * @brief Gets the next available value reference identifier
-	 * @return uint16_t The next value reference ID
+	 * @return ValueRef The next value reference ID
 	 */
-	uint16_t getNextValueRef();
+	ValueRef getNextValueRef();

 private:
 	/**
@@ -183,10 +183,10 @@

 public:
 	uint16_t currentBlockIndex;
-	uint16_t currentOperationIndex;
+	ValueRef currentOperationIndex;
 	std::vector<Block> blocks;
 	std::vector<operation_identifier> returnRefs;
-	uint16_t lastValueRef = 0;
+	ValueRef lastValueRef = 0;
 	std::unordered_map<Snapshot, operation_identifier> globalTagMap;
 	std::unordered_map<Snapshot, operation_identifier> localTagMap;

Index: nautilus/src/nautilus/tracing/phases/SSACreationPhase.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/src/nautilus/tracing/phases/SSACreationPhase.cpp b/nautilus/src/nautilus/tracing/phases/SSACreationPhase.cpp
--- a/nautilus/src/nautilus/tracing/phases/SSACreationPhase.cpp	(revision c068e7a0f1ccc76d6e149cdf3004e83ec5f3c5bb)
+++ b/nautilus/src/nautilus/tracing/phases/SSACreationPhase.cpp	(revision a71bcabed738167def7a175607904934b8ad4bf0)
@@ -169,7 +169,7 @@
 void SSACreationPhase::SSACreationPhaseContext::removeAssignOperations() {
 	// Iterate over all block and eliminate the ASSIGN operation.
 	for (Block& block : trace->getBlocks()) {
-		std::unordered_map<uint16_t, uint16_t> assignmentMap;
+		std::unordered_map<ValueRef, ValueRef> assignmentMap;
 		for (auto& operation : block.operations) {
 			if (operation.op == Op::ASSIGN) {
 				auto& valueRef = get<TypedValueRef>(operation.input[0]);
@@ -210,7 +210,7 @@

 void SSACreationPhase::SSACreationPhaseContext::makeBlockArgumentsUnique() {
 	for (Block& block : trace->getBlocks()) {
-		std::unordered_map<uint16_t, uint16_t> blockArgumentMap;
+		std::unordered_map<ValueRef, ValueRef> blockArgumentMap;

 		// iterate over all arguments of this block and create new ValRefs if the
 		// argument ref is not local. for (uint64_t argIndex = 0; argIndex <
