From 9012a0e0f1a9a15088067adaba2172f8c3304b93 Mon Sep 17 00:00:00 2001
From: lukas schwerdtfeger <lukas.schwerdtfeger@gmail.com>
Date: Wed, 16 Apr 2025 13:06:57 +0200
Subject: [PATCH] disable-mlir-multithreading

---
 .../nautilus/compiler/backends/mlir/MLIRCompilationBackend.cpp  | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/nautilus/src/nautilus/compiler/backends/mlir/MLIRCompilationBackend.cpp b/nautilus/src/nautilus/compiler/backends/mlir/MLIRCompilationBackend.cpp
index 1db6db9..db8a711 100644
--- a/nautilus/src/nautilus/compiler/backends/mlir/MLIRCompilationBackend.cpp
+++ b/nautilus/src/nautilus/compiler/backends/mlir/MLIRCompilationBackend.cpp
@@ -38,6 +38,8 @@ std::unique_ptr<Executable> MLIRCompilationBackend::compile(const std::shared_pt
 	::mlir::LLVM::registerInlinerInterface(registry);
 	::mlir::MLIRContext context(registry);
 
+	context.disableMultithreading();
+
 	auto loweringProvider = std::make_unique<MLIRLoweringProvider>(context);
 	auto mlirModule = loweringProvider->generateModuleFromIR(ir);
 	if (*mlirModule == nullptr) {
-- 
2.45.2

