--- a/nautilus/include/nautilus/val_ptr.hpp
+++ b/nautilus/include/nautilus/val_ptr.hpp
@@ -3,10 +3,12 @@
 
 #include "nautilus/static.hpp"
 #include "nautilus/tracing/TracingUtil.hpp"
+#include "nautilus/tracing/TypedValueRef.hpp"
 #include "nautilus/tracing/Types.hpp"
 #include "nautilus/val.hpp"
 #include "nautilus/val_concepts.hpp"
 #include <cstdint>
+#include <type_traits>
 #include <utility>
 
 namespace nautilus {
@@ -203,12 +205,47 @@
 	ValuePtrType value;
 };
 
+template <typename T, typename F>
+std::size_t field_offset(F T::* pm) {
+	alignas(T) std::byte storage[sizeof(T)] {};
+	T* obj = std::launder(reinterpret_cast<T*>(storage));
+	return reinterpret_cast<char*>(&(obj->*pm)) - reinterpret_cast<char*>(obj);
+}
+
 template <is_ptr ValuePtrType>
 class val<ValuePtrType> : public base_ptr_val<ValuePtrType> {
 public:
 	using base_ptr_val<ValuePtrType>::base_ptr_val;
 	using ValType = typename base_ptr_val<ValuePtrType>::ValType;
 
+	template <typename F, typename T = ValType>
+	    requires std::is_class_v<T>
+	auto get(F T::* pm) {
+		auto offset = field_offset(pm);
+		val<uint8_t*> bytePtr = static_cast<val<uint8_t*>>(*this);
+		val<uint8_t*> fieldBytePtr = bytePtr + offset;
+		val<F*> fieldPtr = static_cast<val<F*>>(fieldBytePtr);
+#ifdef ENABLE_TRACING
+		return val<F&>(fieldPtr, fieldPtr.state);
+#else
+		return val<F&>(fieldPtr);
+#endif
+	}
+
+	template <typename F, typename T = ValType>
+	    requires std::is_class_v<T>
+	void set(F T::* pm, val<F> value) {
+		val<F&> valueRef = get(pm);
+		valueRef = value;
+	}
+
+	template <typename F, typename T = ValType>
+	    requires std::is_class_v<T>
+	void set(F T::* pm, F value) {
+		val<F&> valueRef = get(pm);
+		valueRef = value;
+	}
+
 #ifdef ENABLE_TRACING
 	val(const val<ValuePtrType>& otherValue)
 	    : base_ptr_val<ValuePtrType>(otherValue.value, tracing::traceCopy(otherValue.state)) {
