Subject: [PATCH] 0003-do-not-enable-null-driver-in-debug
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision 4a884e5c57040c8f9b8505d3a286354eeb59a1fd)
+++ b/CMakeLists.txt	(date 1764841555092)
@@ -288,10 +288,6 @@
 # We don't want to ship release builds with it or count it when calculating
 # binary sizes of minified builds.
 set(IREE_HAL_DRIVER_NULL_DEFAULT OFF)
-string(TOUPPER "${CMAKE_BUILD_TYPE}" _UPPERCASE_CMAKE_BUILD_TYPE)
-if(IREE_DEV_MODE OR (_UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "DEBUG"))
-  set(IREE_HAL_DRIVER_NULL_DEFAULT ON)
-endif()
 
 # Vulkan support is enabled by default if the platform might support Vulkan.
 # Apple platforms support Metal instead of Vulkan, though MoltenVK may work.
